### PR TITLE
Feature/router

### PR DESCRIPTION
**Pull request by AndreiBike**

### CODE REVIEW

File: src/client/App.js [hunk 1]
```diff
@@ -1,33 +1,90 @@
...
   netflixRoulette: ["NETFLIX", "ROULETTE"],
   findYourMovie: "FIND YOUR MOVIE",
 };
 
 const footerLabels = {
   netflixRoulette: ["NETFLIX", "ROULETTE"],
-} 
+}
 
+const SortbarContainer = withIdConnection(withConnect(Sortbar));
```
Comment (NurbekAman) [inline src/client/App.js:28]: you can use `compose` from `redux`

File: src/client/App.js [hunk 2]
```diff
@@ -1,33 +1,90 @@
...
+const MoviesContainer = withConnect(Movies);
+const DescriptionContainer = withConnect(withIdConnection(Description));
```
Comment (NurbekAman) [inline src/client/App.js:30]: the same use `compose`

File: src/client/App.test.js [hunk 3]
```diff
@@ -0,0 +1,14 @@
+import React from 'react';
+import {configure, mount} from 'enzyme';
+import Adapter from 'enzyme-adapter-react-16';
+import App from "./App";
+
+configure({adapter: new Adapter()});
+
+describe("<App> component testing", () => {
+  const component = mount(<App />);
```
Comment (NurbekAman) [inline src/client/App.test.js:9]: move it inside `it` test function

File: src/client/components/description/specification/Specification.js [hunk 4]
```diff
@@ -41,13 +42,7 @@ const Specification = (props) => {
...
 
 Specification.propTypes = {
-  id: PropTypes.number,
-  name: PropTypes.string,
-  rating: PropTypes.number,
-  gengre: PropTypes.string,
-  year: PropTypes.number,
-  duration: PropTypes.number,
-  description: PropTypes.string,
+  movie: PropTypes.object,
```
Comment (NurbekAman) [inline src/client/components/description/specification/Specification.js:45]: please always try to describe the shape of object, array please see `PropTypes.arrayof(), PropTypes.objectOf(), PropTypes.shape()`

File: src/client/components/footer/__snapshots__/Footer.test.js.snap [hunk 5]
```diff
@@ -0,0 +1,221 @@
+// Jest Snapshot v1, https://goo.gl/fbAQLP
+
+exports[`<Footer> Snapshot +++capturing snapshot of Footer 1`] = `
```
Comment (NurbekAman) [inline src/client/components/footer/__snapshots__/Footer.test.js.snap:3]: this is not informative. I would see the dom tree, not a react childs

File: src/client/components/footer/Footer.test.js [hunk 6]
```diff
@@ -0,0 +1,20 @@
+import React from 'react';
+import {configure, render} from 'enzyme';
```
Comment (NurbekAman) [inline src/client/components/footer/Footer.test.js:2]: use mount or shallow

File: src/client/components/footer/Footer.test.js [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+import Footer from './Footer';
+
+configure({adapter: new Adapter()});
+
+const footerLabelsTest = {
+  netflixRoulette: ["NETFLIX", "ROULETTE"],
+};
+
+describe("<Footer> Snapshot", () => {
+  const component = render(<Footer footerLabels = {footerLabelsTest} />);
```
Comment (NurbekAman) [inline src/client/components/footer/Footer.test.js:13]: use mount

File: src/client/components/footer/Footer.test.js [hunk 8]
```diff
@@ -0,0 +1,20 @@
...
+  
+  it('+++capturing snapshot of Footer', () => {
+    expect(component).toMatchSnapshot();
```
Comment (NurbekAman) [inline src/client/components/footer/Footer.test.js:16]: try to make `expect(component.toJSON()).toMatchSnapshot()`

File: src/client/components/header/Header.js [hunk 9]
```diff
@@ -19,7 +24,21 @@ export const Header = (props) => {
...
-      <Search />
+      <SearchContainer match = {match}/>
     </div>
   );
-}
\ No newline at end of file
+}
+
+Header.propTypes = {
+  headerLabels: PropTypes.object,
```
Comment (NurbekAman) [inline src/client/components/header/Header.js:33]: You can describe the shape of `headerLabel` by 
`headeLabel: PropTypes.shape({
  netflixRoulette: PropTypes.arrayOf(PropTypes.oneOf(['netflix', 'roulette'])),
 findYourMovie: PropTypes.string
})`

File: src/client/components/header/Header.test.js [hunk 10]
```diff
@@ -0,0 +1,18 @@
...
+
+describe("<Header /> component testing", ()=> {
+  let component, headerLabelsTest;
+  headerLabelsTest = {
+    netflixRoulette: ["NETFLIX", "ROULETTE"],
+    findYourMovie: "FIND YOUR MOVIE",
+  };
+  component = shallow(<Header headerLabels = {headerLabelsTest} />);
+  it("test #1 snapshot testing", ()=>{
+    expect(component).toMatchSnapshot();
```
Comment (NurbekAman) [inline src/client/components/header/Header.test.js:16]: try to use `component.toJSON()`

File: src/client/components/header/search/Search.js [hunk 11]
```diff
@@ -1,62 +1,92 @@
...
-    })
+      'active': this.props.searchBy === 'genres',
+    });
 
     return (
       <div className="search">
         <div className="search-input">
-          <input className="search-input-field" type="search" placeholder="Search" />
-          <button className="search-input-button"> SEARCH </button>
+          <input value={this.props.searchText} onChange={(e) => { this.props.changeSearchText({ searchText: e.target.value }) }} className="search-input-field" type="search" placeholder="Search" />
```
Comment (NurbekAman) [inline src/client/components/header/search/Search.js:66]: `onChange={({ target: { value } }) => this.props.changeSearchText({ value })}`

File: src/client/components/movies/Movies.js [hunk 12]
```diff
@@ -1,58 +1,65 @@
...
-    if (this.state.movies.length) {
-      const renderMovie = (movie) => { return <Movie key={movie.id} movie={movie} /> }
-      let moviesList = this.state.movies.map(renderMovie);
-      return (
+  if (movies.length) {
+    let moviesList = movies.map(renderMovie);
+    return (
+      <InfiniteScroll
+        dataLength={movies.length}
+        next={() => {
```
Comment (NurbekAman) [inline src/client/components/movies/Movies.js:39]: please memoize it

File: src/client/App.js (hunk 1 above)
Comment (AndreiBike) [reply src/client/App.js:28]: Correct)

File: src/client/App.js (hunk 2 above)
Comment (AndreiBike) [reply src/client/App.js:30]: The same

File: src/client/components/footer/Footer.test.js (hunk 8 above)
Comment (AndreiBike) [reply src/client/components/footer/Footer.test.js:16]: Done)

File: src/client/components/header/Header.js (hunk 9 above)
Comment (AndreiBike) [reply src/client/components/header/Header.js:33]: Done)

File: src/client/components/header/Header.test.js (hunk 10 above)
Comment (AndreiBike) [reply src/client/components/header/Header.test.js:16]: Done)

File: src/client/components/footer/Footer.test.js (hunk 7 above)
Comment (AndreiBike) [reply src/client/components/footer/Footer.test.js:13]: I used renderer from 'react-test-renderer'

File: src/client/components/footer/Footer.test.js (hunk 6 above)
Comment (AndreiBike) [reply src/client/components/footer/Footer.test.js:2]: I used renderer from 'react-test-renderer'

File: src/client/components/description/specification/Specification.js (hunk 4 above)
Comment (AndreiBike) [reply src/client/components/description/specification/Specification.js:45]: Done)

File: src/client/components/footer/__snapshots__/Footer.test.js.snap (hunk 5 above)
Comment (AndreiBike) [reply src/client/components/footer/__snapshots__/Footer.test.js.snap:3]: Dom tree created in snapshot

File: src/client/App.test.js (hunk 3 above)
Comment (AndreiBike) [reply src/client/App.test.js:9]: Corrected

**Pull request closed without merging**